tests/Kiota.Builder.Tests/KiotaBuilderTests.cs

```diff
@@ -4677,20 +4679,35 @@ public void ModelsUseDescriptionWhenAvailable()
             }
         };
         var mockLogger = new Mock<ILogger<KiotaBuilder>>();
-        var builder = new KiotaBuilder(mockLogger.Object, new GenerationConfiguration { ClientClassName = "TestClient", ClientNamespaceName = "TestSdk", ApiRootUrl = "https://localhost" }, _httpClient);
+        var builder = new KiotaBuilder(mockLogger.Object, new GenerationConfiguration { ClientClassName = "TestClient", ClientNamespaceName = "TestSdk", ApiRootUrl = "https://localhost", ExcludeBackwardCompatible = excludeBackwardCompatible }, _httpClient);
         var node = builder.CreateUriSpace(document);
         var codeModel = builder.CreateSourceModel(node);
         var modelsSubNS = codeModel.FindNamespaceByName("TestSdk.answer");
         Assert.NotNull(modelsSubNS);
-        var responseClass = modelsSubNS.Classes.FirstOrDefault(x => x.IsOfKind(CodeClassKind.Model));
+        var responseClass = modelsSubNS.FindChildByName<CodeClass>("AnswerGetResponse", false);
         Assert.NotNull(responseClass);
         Assert.Equal("some description", responseClass.Documentation.Description);
 
-        responseClass = modelsSubNS.Classes.FirstOrDefault(c => c.IsOfKind(CodeClassKind.RequestBuilder));
-        Assert.NotNull(responseClass);
-        Assert.Equal("some path item description", responseClass.Documentation.Description);
+        var obsoleteResponseClass = modelsSubNS.FindChildByName<CodeClass>("AnswerResponse", false);
+        if (excludeBackwardCompatible)
+            Assert.Null(obsoleteResponseClass);
+        else
+        {
+            Assert.NotNull(obsoleteResponseClass);
+            Assert.Equal("some description", obsoleteResponseClass.Documentation.Description);
```

Comment:
In the case where a path has only one operation and `excludeBackwardCompatible` is not enabled, does it make sense to not generate the name disambiguated as there won't be another type colliding and consequently not generate the deprecation information. 
In my head, in such a scenario, we will be adding methods/models in a scenario that doesn't need the collission to the resolved.
When a new path is added the code should now add the disambiguated models/methods and then deprecate the single one that existed before as a way to manage generation of extra code that may not really be used.

Reply:
this suggestion makes for a moving target which is going to degrade the customer experience in my opinion.

 # current implementation

| exclude backward compatible | # of operations with inline types | name for the non obsolete Get inline type |
| ------------ | ------------- | --------------- |
| true | 2 | SegmentGetResponse | 
| true | 1 | SegmentGetResponse | 
| false | 2 | SegmentGetResponse |
| false | 1 | SegmentGetResponse | 

# suggested implementation

| exclude backward compatible | # of operations with inline types | name for the non obsolete Get inline type |
| ------------ | ------------- | --------------- |
| true | 2 | SegmentGetResponse | 
| true | 1 | SegmentResponse | 
| false | 2 | SegmentGetResponse |
| false | 1 | SegmentResponse | 

See how with this suggestion the type name varies based on the description? making a description change potentially breaking for the API consumer when we don't need to?

Reply:
Thanks for taking the time to explain this. I believe you're right there. 
The generation of different model types based on the setting would be degraded customer experience given the effort here  to be backward compatible.